### ISSUE TITLE
OSError: [Errno 9] Bad file descriptor

### ISSUE THREAD

**Issue by juancrescente** [celery/celery#5517]
This one seems old but I cannot find any clues on how to solve it.

```
$ celery --version                         
4.3.0 (rhubarb)
```

```

[2019-05-11 19:01:35,734: INFO/MainProcess] Received task: chat.tasks.execute[7afeda12-c5dd-4214-b402-78d3e8ead98e]  
[2019-05-11 19:01:37,485: WARNING/ForkPoolWorker-2] [['Symbol', 'Date', 'Time', 'Open', 'High', 'Low', 'Close', 'Volume'], ['AAPL.US', '2019-05-10', '22:00:32', '197.419', '198.85', '192.77', '197.18', '41208712']]
[2019-05-11 19:01:37,494: ERROR/ForkPoolWorker-2] Task chat.tasks.execute[7afeda12-c5dd-4214-b402-78d3e8ead98e] raised unexpected: OSError(9, 'Bad file descriptor')
Traceback (most recent call last):
  File "/Users/juan/Documents/manu/dev/python_challenge/venv/lib/python3.6/site-packages/celery/app/trace.py", line 385, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/Users/juan/Documents/manu/dev/python_challenge/venv/lib/python3.6/site-packages/celery/app/trace.py", line 648, in __protected_call__
    return self.run(*args, **kwargs)
  File "/Users/juan/Documents/manu/dev/python_challenge/chat/tasks.py", line 14, in execute
    loop.run_until_complete(sendData(stock(parameter), "BOT", room_group_name))
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/asyncio/base_events.py", line 455, in run_until_complete
    self.run_forever()
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/asyncio/base_events.py", line 422, in run_forever
    self._run_once()
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/asyncio/base_events.py", line 1396, in _run_once
    event_list = self._selector.select(timeout)
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/selectors.py", line 577, in select
    kev_list = self._kqueue.control(None, max_ev, timeout)
OSError: [Errno 9] Bad file descriptor
^C

```

**Comment by auvipy** [comment]
closing as duplicate. comment on older issues that are related

**Comment by troelsfr** [comment]
Could you please link to the duplicate error report?

**Comment by troelsfr** [comment]
Nevermind. Solved the problem by upgrading to v4.4.7 :)
